src/pages/Feed/components/FeedModal/FeedModal.tsx

```diff
@@ -0,0 +1,87 @@
+import SelectModal from '../../../../components/Modal/SelectModal/SelectModal';
+import OKModal from '../../../../components/Modal/OKModal/OKModal';
+import FriendModal from '../../../../components/Modal/FriendModal/FriendModal';
+import useDeleteFriend from '../../hooks/queries/useDeleteFriend';
+import usePatchFix from '../../hooks/queries/usePatchFix';
+import { FriendType, ModalType, ActionType } from '../../types';
+import React, { useState } from 'react';
+
+interface FeedModalProps {
+  currentFriend: FriendType;
+  setCurrentFriend: React.Dispatch<React.SetStateAction<FriendType>>;
+  closeModal: () => void;
+  setAction: React.Dispatch<React.SetStateAction<ActionType>>;
+}
+
+const FeedModal = ({
+  currentFriend,
+  setCurrentFriend,
+  closeModal,
+  setAction,
+}: FeedModalProps) => {
+  const [modalType, setModalType] = useState<ModalType>('friend');
+  const { mutate: deleteFriend } = useDeleteFriend();
+  const { mutate: patchFix } = usePatchFix();
+
+  const handleDelete = () => {
+    deleteFriend(currentFriend.userID, {
+      onSuccess: () => {
+        setAction('delete');
+        setModalType('ok');
+      },
+    });
+  };
+
+  const handleFix = () => {
+    patchFix(currentFriend.userID, {
+      onSuccess: () => {
+        setAction('fix');
+        setCurrentFriend((prev: FriendType) => ({
+          ...prev,
+          isFixed: !prev.isFixed,
+        }));
+      },
+    });
+  };
+
+  const handleClose = () => {
+    closeModal();
+    setModalType('friend');
+    setAction(null);
+  };
+
+  switch (modalType) {
+    case 'delete':
+      return (
+        <SelectModal
+          type="delete"
+          content="이 행위는 되돌릴 수 없습니다."
+          onClose={handleClose}
+          onSubmit={handleDelete}
+        >
+          <span style={{ color: '#FAED46' }}>{currentFriend?.nickname}</span>
+          님을 삭제하겠습니까?
+        </SelectModal>
+      );
+    case 'ok':
+      return (
+        <OKModal onClose={handleClose}>
+          친구 관계를 <br /> 성공적으로 삭제했습니다.
+        </OKModal>
```

Comment:
사용자가 OK 버튼을 누를 때 직접 상태를 리셋

```suggestion
        <OKModal
          onClose={() => {
            setCurrentFriend(null);
            handleClose();
          }}
        >
          친구 관계를 <br /> 성공적으로 삭제했습니다.
        </OKModal>
```

Reply:
의견 감사합니다 !

처음에는 해당 방식을 위주로 수정을 진행했었는데, 삭제 이후 확인 버튼을 누르는 동작을 해야만 currentFriend 가 반영이 됩니다..
개인적으로 생각했을 때 삭제를 하자마자 삭제된 리스트가 반영이 되지 않으면 삭제가 되었나? 라는 의문이 들 수 있을 것 같으며, 확인 버튼이 아닌 모달 상단 닫기 버튼을 누를 경우에도 상태가 갱신이 되지 않을 수 있더라구요. 

개인적인 생각이었지만, 만약 사용자 경험 측면에서 큰 문제가 없을 것 같다며 해당 방식을 적용하는 것이 더 나은 것 같습니다!